### PR TITLE
Quickstart django astro

### PR DESCRIPTION
**Pull request by sixhobbits**
Depends on https://github.com/ritza-co/fusionauth-quickstart-python-django-web being added to `fusionauth` org

~~TODO - still needs a github action to test~~ [edit done ✅]

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205273462533540
  - https://app.asana.com/0/0/1205271148607511

### CODE REVIEW

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+```
+
+Now you will work from the top down to set up authentication URLs in the project `settings.py`, then in the application `urls.py`, then the `views.py`, and finally, the HTML templates for the views.
+
+
+### Project Settings
+
+Overwrite your `mysite/settings.py` contents with the following code.
+
+`settings.py`
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:154]: Same issue here with the filename hanging out without any context.

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+<RemoteCode url="https://raw.githubusercontent.com/FusionAuth/fusionauth-quickstart-python-django-web/main/complete-application/mysite/mysite/settings.py" lang="python" />
+
+Here, your `app` and `mozilla_django_oidc` are added to the list of `INSTALLED_APPS`. The authentication library from Mozilla is set up by `mozilla_django_oidc` and the `OIDC_OP_` constants imported from the `.env` file. This automatically handles the interaction with FusionAuth on the endpoints configured in this file and the matching ones configured in the `kickstart.json` file that you used to initialize FusionAuth.
+
+### Project URLs
+
+Overwrite the `mysite/urls.py` file with the following code.
+
+`urls.py`
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:163]: Same issue here with the filename hanging out without any context.

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+ * The `app` URLs are for the app you created in your project.
+ * The `oidc` URLs are automatically generated from the settings you configured in the previous step.
+
+### App URLs
+
+From this point, you'll work in the `app` not in the main project. 
+
+Create an `app/urls.py` file and set its content to match the project URLs.
+
+`urls.py`
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:177]: Same issue here with the filename hanging out without any context.

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+<RemoteCode url="https://raw.githubusercontent.com/FusionAuth/fusionauth-quickstart-python-django-web/main/complete-application/mysite/app/urls.py"
+ lang="python" />
+
+### App Views
+
+Now overwrite the `app/views.py` file. There are three pages: the home page, the account page, and the make change page.
+
+`views.py`
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:185]: Same issue here with the filename hanging out without any context.

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+* [User data and custom claims in access token JWTs](https://fusionauth.io/articles/tokens/jwt-components-explained).
+
+### Security
+
+* You may want to customize the [token expiration times and policies](https://fusionauth.io/docs/v1/tech/oauth/#configure-application-oauth-settings) in FusionAuth.
+* Choose [password rules](https://fusionauth.io/docs/v1/tech/core-concepts/tenants#password) and a [hashing algorithm](https://fusionauth.io/docs/v1/tech/reference/password-hashes) that meet your security needs.
+
+### Tenant and Application Management
+
+* Model your application topology using [Applications](https://fusionauth.io/docs/v1/tech/core-concepts/applications), [Roles](https://fusionauth.io/docs/v1/tech/core-concepts/roles), [Groups](https://fusionauth.io/docs/v1/tech/core-concepts/groups), [Entities](https://fusionauth.io/docs/v1/tech/core-concepts/groups), and more.
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:256]: Please make all links to the FusionAuth site relative.

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+<RemoteCode url="https://raw.githubusercontent.com/FusionAuth/fusionauth-quickstart-python-django-web/main/complete-application/mysite/mysite/urls.py"
+ lang="python" />
+
+ * The `admin` site is the default Django admin.
+ * The `app` URLs are for the app you created in your project.
+ * The `oidc` URLs are automatically generated from the settings you configured in the previous step.
+
+### App URLs
+
+From this point, you'll work in the `app` not in the main project.
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:173]: Should you `cd` into that directory, then?

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+* The `app` view, or home page, redirects the user to the `account` view if they are already logged in.
+* The `account` view checks whether the user is logged in. If the user is logged in, it returns the account page template, passing in the user's email address returned from FusionAuth. If the user isn't logged in, they are redirected to the login page. This also happens in the change page.
+* The `logout` view is necessary because you need to clear the session for the user in Django after the user logs out of FusionAuth. FusionAuth sends the user to this view after logging them out, which then returns them to the home page.
+* The `change` view looks longer only because it demonstrates how to do some calculations if the request is a POST instead of a GET. These calculations are then passed to the template in the `change` variable to display onscreen.
+
+## Customization
+
+The final step is to create the HTML pages and style them with CSS.
+
+* Copy the `static` folder with its contents from `complete-application/mysite/app/static` to your `/mysite/app/static` location.
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:200]: Please provide commands for me to copy. Maybe state where you are running the commands from.

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+* Copy the `templates` folder with its contents from `complete-application/mysite/app/templates` to your `/mysite/app/templates` location.
+
+There are only two lines in these templates related to authentication. In the home template, there is a login button.
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:203]:
```suggestion
There are only two lines in these templates related to authentication. In the `home.html` template, there is a login button.
```

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+
+In the account and change pages, there is a logout form.
+
+```html
+<form id="logoutForm" class="button-lg"  action="{% url 'oidc_logout' %}" method="post">
+  {% csrf_token %}
+  <a class="button-lg" href="#" onclick="document.getElementById('logoutForm').submit();">Log out</a>
+</form>
+```
+Log out has to be a form that submits a POST request to FusionAuth. Here we use an anchor element that is easy to style, but if your users might disable JavaScript, you could use a traditional form `input` element.
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:218]:
```suggestion

Log out has to be a form that submits a `POST` request to FusionAuth. Here we use an anchor element that is easy to style, but if your users might disable JavaScript, you could use a traditional form `input` element.
```

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+
+## Run the Application
+
+Update the default SQLite database before testing.
+
+```bash
+python manage.py migrate
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:225]: where do I need to be to run `manage.py`? I guess we didn't tell folks to `cd` anywhere, so I'm still supposed to be `fusionauth-quickstart-python-django-web/mysite`.

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+```
+
+You can now browse to [the Django app](http://localhost:8000/app).
+
+If the app is not running, you can use the following command.
+
+```bash
+source venv/bin/activate && python mysite/manage.py runserver
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:233]: Would I do this in a new window? Why do I need the `venv` command? Where should I run this from?

Didn't I start the server up above and you haven't told me to stop it? I'm confused.

File: site/docs/v1/tech/tutorials/integrate-python-django.adoc
Comment (mooreds) [inline site/docs/v1/tech/tutorials/integrate-python-django.adoc:1]: Should remove this file entirely.

**Review by mooreds — CHANGES_REQUESTED** [review]
* I saw a we or two, please use you. 

* Why do we need CORS open to port 5000 in the example app in the kickstart?

* remove the django guide example from the example apps data file (`site/_data/exampleapps.yml)

* add a task to archive the guide example to the GH PR (I can do it, just want a task in the PR desc)

* add a redirect to the cloudfront redirector (since this was previously a doc tutorial)

edit src/cloudfront/fusionauth-website-request-handler.js and look for the rails quickstart line:

rd[d+'/tutorials/integrate-ruby-rails']= '/docs/quickstarts/quickstart-ruby-rails-web';

add a similar one for the python django quickstart

File: src/cloudfront/fusionauth-website-request-handler.js [hunk 12]
```diff
@@ -91,7 +91,8 @@ rd[d+'/tutorials/migrate-users']=d+'/migration-guide/tutorial';
 rd[d+'/tutorials/setting-up-user-account-lockout']=d+'/tutorials/gating/setting-up-user-account-lockout';
 rd[d+'/tutorials/two-factor/authenticator-app']=d+'/tutorials/two-factor/authenticator-app-pre-1-26';
 rd[d+'/tutorials/two-factor/twilio-push']=d+'/tutorials/two-factor/twilio-push-pre-1-26';
-rd[d+'/tutorials/integrate-python-flask']= '/docs/quickstarts/quickstart-python-flask-web';
+rd[d+'/tutorials/integrate-python-django']= '/docs/quickstarts/quickstart-python-django-web';
+rd[d+'/tutorials/integrate-python-django']= '/docs/quickstarts/quickstart-python-flask-web';
```
Comment (mooreds) [inline src/cloudfront/fusionauth-website-request-handler.js:95]:
```suggestion
rd[d+'/tutorials/integrate-python-flask']= '/docs/quickstarts/quickstart-python-flask-web';
```

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+
+### FusionAuth Customization
+
+FusionAuth gives you the ability to customize just about everything to do with the user's experience and the integration of your application. This includes:
+* [Hosted pages](/docs/v1/tech/themes/) such as login, registration, email verification, and many more.
+* [Email templates](/docs/v1/tech/email-templates/email-templates).
+* [User data and custom claims in access token JWTs](/articles/tokens/jwt-components-explained).
+
+### Security
+
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:241]:
```suggestion

* Learn more about configuring the [open source OIDC library](https://github.com/mozilla/mozilla-django-oidc).
```

**Review by mooreds — APPROVED** [review]
2 minor changes. Need to put this through SEO review, but other than that, this is good to go.

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx [hunk 14]
```diff
@@ -0,0 +1,250 @@
+---
+title: Python Django
```
Comment (bradmccarty) [inline astro/src/content/quickstarts/quickstart-python-django-web.mdx:2]: Is this title intentional? If not, I'd prefer we change to "Adding Authentication To A Python And Django App With FusionAuth"

**Review by bradmccarty — COMMENTED** [review]
@mooreds My only suggestion is RE: the title. Left a comment inline.

File: astro/src/content/quickstarts/quickstart-python-django-web.mdx (hunk 14 above)
Comment (mooreds) [reply astro/src/content/quickstarts/quickstart-python-django-web.mdx:2]: @bradmccarty the title of all the quickstarts is just the technology. 

https://fusionauth.io/docs/quickstarts/quickstart-python-flask-web

It is also what shows up on the quickstarts index:

https://fusionauth.io/docs/quickstarts/

I could switch this around for all of the quickstarts if you think it is very important.

Comment (bradmccarty) [reply astro/src/content/quickstarts/quickstart-python-django-web.mdx:2]: @mooreds Nope. Not that important. We're ranking well for all of those keywords anyway. Thanks for the clarification.

**Pull request merged**